src/pages/learning-journal.tsx

```diff
@@ -32,129 +50,192 @@ export async function getStaticProps () {
   }
 }
 
-type FormattedLearningJournal = Omit<LearningJournalType, 'date'> & {
-  dateTitle: string;
-}
+function LearningJournalContent () {
+  const [date, setDate] = useState<string>()
+  const [afterCursor, setAfterCursor] = useState<string>()
+  const [beforeCursor, setBeforeCursor] = useState<string>()
+
+  const { data: { edges, pageInfo } = {}, isValidating } = useSWR(
+    ['learning-journal-page', date, afterCursor, beforeCursor],
+    () => getLearningJournalPage({ date, after: afterCursor, before: beforeCursor })
+  )
 
-export default function LearningJournal ({
-  learningJournals
-}: InferGetStaticPropsType<typeof getStaticProps>) {
-  const formattedLearningJournals = learningJournals.map(({ date, ...rest }) => ({
+  const formattedEntries = edges?.map(({ node: { date, ...rest } }) => ({
     ...rest,
-    dateTitle: DateTime.fromISO(date)
-      .toFormat('DD')
+    dateTitle: DateTime.fromISO(date).toFormat('DD')
   }))
 
-  const {
-    data,
-    hasMoreItems,
-    handlePagination
-  } = usePagination<FormattedLearningJournal>({
-    list: formattedLearningJournals,
-    itemsPerPage: 2
-  })
+  const handleInputValueChange = (inputValue: string) => {
+    setDate(inputValue)
+  }
+
+  const { hasPreviousPage, hasNextPage, startCursor, endCursor } = pageInfo ?? {}
+
+  const handleNextPage = () => {
+    setAfterCursor(endCursor)
+    setBeforeCursor(undefined)
+  }
+
+  const handlePrevPage = () => {
+    setAfterCursor(undefined)
+    setBeforeCursor(startCursor)
+  }
+
+  const hasEntries = formattedEntries?.length > 0 && !isValidating
```

Comment:
Rename this flag since it's now also taking into consideration `isValidation`, so it's broader than just a variable length check 

I would suggest renaming to `shouldShowEntries`